src/DotNetty.Common/Concurrency/TaskCompletionSource.cs

```diff
@@ -34,6 +34,11 @@ public bool setUncancellable()
             return true;
         }
 
+        public override string ToString()
+        {
+            return "TaskCompletionSource[status: " + this.Task.Status.ToString() + "]";
```

Comment:
it's `Concat(string..)` vs `Concat(object..)`. I prefer to explicitly call `ToString` to avoid hidden boxing. In this case though, it'll get boxed anyway (`Enum.ToString()` issue) but this is for debugging / unhappy path logging only anyway.